system/lib/wasmfs/syscalls.cpp

```diff
@@ -38,7 +38,7 @@ long __syscall_dup2(long oldfd, long newfd) {
   }
 
   if (oldfd == newfd) {
-    return oldfd;
+    return -EINVAL;
```

Comment:
Is this a bugfix?

Reply:
No, this is because we switch from implementing __syscall_dup2 to __syscall_dup3 here and this is one of the differences between the two.